### ISSUE TITLE
ClearML no auth training ValueError bug

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/yolov5#9877]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

@thepycoder ClearML raises ValueError when installed but not authenticated.

<img width="788" alt="Screenshot 2022-10-20 at 20 23 40" src="https://user-images.githubusercontent.com/26833433/197028162-abc087a1-5ac6-4ee9-b92f-f992c08adb38.png">

### Environment

Python 3.7
Ubuntu
Notebook

### Minimal Reproducible Example

```ipython
!git clone https://github.com/ultralytics/yolov5  # clone
%cd yolov5
%pip install -qr requirements.txt clearml # install

!python train.py
```

### Additional

Should probably handle as a console printout/warning, i.e. `ClearML: installed but not logged in. Please login to log training.`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by thepycoder** [comment]
Hi @glenn-jocher,

You're right, this is not the best experience, We're working right now to fix this for clearml in general, the change will then propagate to YOLOv5 as well. How urgent is this and are you ok with it being a fix on our end (meaning earlier package versions will still get it) or would you prefer a separate fix in the YOLOv5 repo as well?

**Comment by glenn-jocher** [comment]
@thepycoder yes we should sync up to see how we can streamline onboarding, in particular for the Colab notebook. I think there's room for improvement.

**Comment by creatorcaty** [comment]
Same issue

**Comment by pderrenger** [comment]
Yep—this occurs when `clearml` is installed but not configured; running `clearml-init` to create your ClearML credentials resolves it (see the [Ultralytics ClearML tracking guide](https://www.ultralytics.com/blog/remotely-train-and-monitor-yolov5-using-clearml)), and if you don’t want ClearML logging you can simply `pip uninstall clearml`; if it still fails on the latest YOLOv5 `master` + latest `clearml`, please paste the full traceback and your `pip show clearml` output.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #13809](https://github.com/ultralytics/yolov5/pull/13809) by @glenn-jocher.

Key changes that may resolve this issue:

- Added a dedicated `ClearmlNotConfiguredError` for missing or incomplete ClearML setup.
- Detects expected ClearML configuration problems such as missing access keys, secret keys, host, or credentials.
- Updated logger initialization to suppress only expected “ClearML not configured” cases.
- Preserves the existing user-friendly warning when ClearML is installed but not configured.
- Allows unexpected ClearML errors to surface normally instead of being silently ignored, improving debugging visibility.

Credit to @glenn-jocher and all contributors involved in the PR.

To test whether the fix resolves this issue, install the latest YOLOv5 code from `main`:

```bash
pip install git+https://github.com/ultralytics/yolov5.git@main
```

Alternatively, wait for the next official PyPI release that includes this change.

Please let us know whether the merged changes resolve the issue in your environment. Thanks 🙏 for reporting this, and any further feedback is welcome if the issue persists.
